group/assignment5/Assignment 5.java

```diff
@@ -0,0 +1,257 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Node {
+
+    private char value;
+    private List<Node> children;
+    //keeping track of how many nodes this one depends on
+    private int numOfParents; 
+
+    public Node(char v) {
+        value = v;
+        children = new ArrayList<Node>();
+    }
+
+    public char getValue() {
+        return value;
+    }
+
+    public void addChild(Node n) {
+        children.add(n);
+    }
+
+    public void removeChild(Node n) {
+        children.remove(n);
+    }
+    public void removeChildren() {
+        children = new ArrayList<Node>();
+    }
+
+    public List<Node> getChildren() {
+        return children;
+    }
+
+    public void incNumOfParents() {
+        numOfParents++;
+    }
+
+    public void decNumOfParents() {
+        numOfParents--;
+    }
+
+    public int getNumOfParents() {
+        return numOfParents;
+    }
+}
+
+public class A5Q1 {
+
+    private static List<Character> buildOneWordSolution(String word) {
+        //one word does not tell us anything about the order od chars
+        //so we simply return all letters
+        //using hashset so that there are no repetition of chars
+        HashSet<Character> result = new HashSet<>();
+        for (int i = 0; i < word.length(); i++) {
+            result.add(word.charAt(i));
+        }
+        LinkedList<Character> listResult = new LinkedList<>(result);
+        return listResult;
+    }
+
+    private static void compareWords(String prevWord, String curWord, Node root, HashMap<Character, Node> visitedChars) {
+
+        boolean differenceFound = false;
+

```

Comment:
I always try to use Maths.min(number1, number2) for these things, I think it's cleaner and easier to do so.